tests/Simplify/EqualTest.elm

```diff
@@ -934,6 +934,787 @@ a = ({ a = 1 }).a == ({ a = 1 }).b
+                                , "Set.isEmpty runs in constant time."
+                                ]
+                            , under = "0 == Set.size s"
+                            }
+                            |> Review.Test.whenFixed """module A exposing (..)
+a : Bool
+a = Set.isEmpty s
+"""
+                        ]
+        , test "should replace (s |> Set.size) == 0 with Set.isEmpty s" <|
+            \() ->
+                """module A exposing (..)
+a : Bool
+a = (s |> Set.size) == 0
+"""
+                    |> Review.Test.run ruleWithDefaults
+                    |> Review.Test.expectErrors
+                        [ Review.Test.error
+                            { message = "This can be replaced with a call to Set.isEmpty"
+                            , details =
+                                [ "Whereas Set.size takes as long to run as the number of elements in the Set,"
+                                , "Set.isEmpty runs in constant time."
+                                ]
+                            , under = "(s |> Set.size) == 0"
+                            }
+                            |> Review.Test.whenFixed """module A exposing (..)
+a : Bool
+a = (s |> Set.isEmpty)
+"""
+                        ]
+        , test "should replace 0 == ([] |> Set.size) with Set.isEmpty s" <|
+            \() ->
+                """module A exposing (..)
+a : Bool
+a = 0 == (s |> Set.size)
+"""
+                    |> Review.Test.run ruleWithDefaults
+                    |> Review.Test.expectErrors
+                        [ Review.Test.error
+                            { message = "This can be replaced with a call to Set.isEmpty"
+                            , details =
+                                [ "Whereas Set.size takes as long to run as the number of elements in the Set,"
+                                , "Set.isEmpty runs in constant time."
+                                ]
+                            , under = "0 == (s |> Set.size)"
+                            }
+                            |> Review.Test.whenFixed """module A exposing (..)
+a : Bool
+a = (s |> Set.isEmpty)
+"""
+                        ]
+        , test "should replace Set.size s /= 0 with (not << Set.isEmpty)" <|
+            \() ->
+                """module A exposing (..)
+a : Bool
+a = Set.size s /= 0
+"""
+                    |> Review.Test.run ruleWithDefaults
+                    |> Review.Test.expectErrors
+                        [ Review.Test.error
+                            { message = "This can be replaced with a call to (not << Set.isEmpty)"
+                            , details =
+                                [ "Whereas Set.size takes as long to run as the number of elements in the Set,"
+                                , "Set.isEmpty runs in constant time."
+                                ]
+                            , under = "Set.size s /= 0"
```

Comment:
I think the range should be put on the smallest section of the code that makes sense. Here, `s` could potentially be a large expression, on multiple lines.

I don't know exactly what would be best.
It could be only under `/= 0`. Though if this is on multiple lines, then it should only be only under `/=`.
But we could also put it under `Set.size`, and - unless someone has a stronger opinion on this - I think it should be there.